src/app/onion/learning-object-builder/components/user-dropdown/user-dropdown.component.ts

```diff
@@ -0,0 +1,142 @@
+import { Component, OnInit, OnDestroy, Output, EventEmitter, Input, IterableDiffers, IterableDiffer, DoCheck } from '@angular/core';
+import { UserService } from 'app/core/user.service';
+import { AuthService } from 'app/core/auth.service';
+import { User } from '@cyber4all/clark-entity';
+import { Subject } from 'rxjs';
+import { takeUntil, debounceTime } from 'rxjs/operators';
+
+@Component({
+  selector: 'clark-user-dropdown',
+  templateUrl: './user-dropdown.component.html',
+  styleUrls: ['./user-dropdown.component.scss'],
+})
+export class UserDropdownComponent implements OnInit, DoCheck, OnDestroy {
+  // array of usernames representing all selected users
+  selectedAuthors: string[] = [];
+  // array of User objects returned from service
+  searchResults: User[] = [];
+
+  // search query string, modeled to the search
+  query: string;
+
+  // true if dropdown results should be shown, false if they should be hidden
+  showDropdown: boolean;
+  // true if the component is actively querying the services, false otherwise
+  loading: boolean;
+
+  // fires every time an input event occurs on the search input element
+  userSearchInput$: Subject<string> = new Subject();
+  // fires when this component is destroyed
+  destroyed$: Subject<void> = new Subject();
+
+  @Input() contributors: User[];
+  differ: IterableDiffer<User>;
+
+  // fired when an author is added as a contributor
+  @Output() addAuthor: EventEmitter<User> = new EventEmitter();
+  // fired when an author is removed as a contributor
+  @Output() removeAuthor: EventEmitter<User> = new EventEmitter();
+
+  constructor(private userService: UserService, private authService: AuthService, private differs: IterableDiffers) {
+    // init contributors iterable differ
+    this.differ = this.differs.find([]).create(null);
+  }
+
+  ngOnInit() {
```

Comment:
Something like this might work here (though you may want to use a name other than `o$`)
```suggestion
 ngOnInit() {
  // Subscribe to the search input and set it to unsubscribe when the component is destroyed
  const o$ = this.userSearchInput$.pipe(takeUntil(this.destroyed$));

  // Fire user search after debounce
  o$.pipe(
    debounceTime(650)
  ).subscribe((val: string) => {
    this.findUser(val.trim());
  });
  // Immediately (without debounce) show the popup
  o$.subscribe((val: string) => {
    if (val && val !== '') {
      this.showDropdown = true;
      this.loading = true;
    } else {
      this.clearSearch();
      this.showDropdown = false;
      this.loading = false;
    }
  });
}
```